server/src/main/java/org/opensearch/rest/action/admin/cluster/RestWlmStatsAction.java

```diff
@@ -53,6 +65,105 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
         Set<String> queryGroupIds = Strings.tokenizeByCommaToSet(request.param("queryGroupId", "_all"));
         Boolean breach = request.hasParam("breach") ? Boolean.parseBoolean(request.param("boolean")) : null;
         WlmStatsRequest wlmStatsRequest = new WlmStatsRequest(nodesIds, queryGroupIds, breach);
+
+        int pageSize = request.paramAsInt("size", 10);  // Default to 10 results per page
+        String nextToken = request.param("next_token");
+        String sortBy = request.param("sort", "node_id"); // Default: sort by node_id
+        String sortOrder = request.param("order", "asc"); // Default: ascending order
+
+        if (!sortOrder.equals("asc") && !sortOrder.equals("desc")) {
+            throw new IllegalArgumentException("Invalid value for 'order'. Allowed: 'asc', 'desc'.");
+        }
+
+        if (!sortBy.equals("node_id") && !sortBy.equals("query_group")) {
+            throw new IllegalArgumentException("Invalid value for 'sort'. Allowed: 'node_id', 'query_group_id'.");
+        }
+
+        boolean isTabular = request.rawPath().contains("_list/wlm_stats");
+
+        if (isTabular) {
+            return channel -> client.admin().cluster().wlmStats(wlmStatsRequest,
+                new RestResponseListener<WlmStatsResponse>(channel) {
+                    @Override
+                    public RestResponse buildResponse(WlmStatsResponse response) throws Exception {
+                        WlmPaginationStrategy paginationStrategy = new WlmPaginationStrategy(pageSize, nextToken, sortBy, sortOrder, response);
+                        List<WlmStats> paginatedStats = paginationStrategy.getPaginatedStats();
+                        Table paginatedTable = buildTable(paginatedStats, paginationStrategy);
+                        PageToken nextPageToken = paginationStrategy.getNextToken();
+
+                        if (nextPageToken != null) {
+                            paginatedTable.setPageToken(nextPageToken);
+                        }
```

Comment:
The nextPageToken should be generated to inject into `Table` instance and should be used inside buildTable method here.